### PR TITLE
Adding filter method

### PR DESCRIPTION
**Pull request by hirokiosame**
This is a continuation of https://github.com/postcss/postcss-import/pull/259 since it seems to be abandoned. I have updated the name to `filter`.

If the logic/implementation looks good, I will go ahead and add documentation.

Example postcss config file:
```
module.exports = {
	plugins: [
		require('postcss-import')({
			filter(url) {
				if (url.match(/font\.css$/)) {
					return false;
				}
			},
		}),
	],
};
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -140,6 +140,14 @@ function parseStyles(result, styles, options, state, media) {
             return
           }
 
+          if (options.filter) {
+            let filtered = options.filter(stmt.uri);
+
+            if (filtered === false) {
+              return;
+            }
```
Comment (RyanZim) [inline index.js]: Can be shortened to `if (!options.filter(stmt.uri)) return`

**Comment by RyanZim** [comment]
Needs tests, not 100% sure that this will work correctly, will have to do a bit of testing myself.

File: index.js (hunk 1 above)
Comment (hirokiosame) [inline index.js]: I do a strict `false` check so blocking is deliberate. When the filter has no return-statement and the return value is `undefined`, the filter will have no effect.

I can shorten it to `if (options.filter(stmt.uri) === false)` if you want but I typically opt for the verbose option as readability -> maintainability especially in collaborative projects.

**Comment by hirokiosame** [comment]
@RyanZim What cases are you suspicious of/interested in handling? I'll try to get around to adding tests this weekend.

**Comment by RyanZim** [comment]
@hirokiosame Ping on the tests?

**Comment by 0xcaff** [comment]
It looks like this one is dead too. Can I pick this PR up?

**Comment by RyanZim** [comment]
Closed by #327

**Pull request closed without merging**